src/main/java/io/jenkins/plugins/orka/helpers/OrkaClientProxy.java

```diff
@@ -0,0 +1,90 @@
+package io.jenkins.plugins.orka.helpers;
+
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import hudson.util.Secret;
+
+import io.jenkins.plugins.orka.client.ConfigurationResponse;
+import io.jenkins.plugins.orka.client.DeletionResponse;
+import io.jenkins.plugins.orka.client.DeploymentResponse;
+import io.jenkins.plugins.orka.client.NodeResponse;
+import io.jenkins.plugins.orka.client.OrkaClient;
+import io.jenkins.plugins.orka.client.VMResponse;
+
+import java.io.IOException;
+import java.util.List;
+
+public class OrkaClientProxy {
+
+    private StandardUsernamePasswordCredentials credentials;
+    private String endpoint;
+
+    public OrkaClientProxy() {
+    }
+
+    public OrkaClientProxy(String endpoint, String credentialsId) {
+        this.setData(endpoint, credentialsId);
+    }
+
+    public void setData(String endpoint, String credentialsId) {
+        this.credentials = CredentialsHelper.lookupSystemCredentials(credentialsId);
```

Comment:
I'm not a fan of this. Makes the code not thread-safe (it can be called from different places/threads) and someone can change the credentials under the nose of someone else.  
What I can suggest is - either remove this and use a new instance of the proxy each time or change the signature of the methods to accept endpoint and credentials. This way you ensure they will not change.  

Additionally, we try to use `this` for fields (it's ok here, but in the methods below it is not used).